src/Types.cc

```diff
@@ -88,5 +88,59 @@ std::ostream &operator<<(std::ostream &_out, const sdf::Errors &_errs)
   }
   return _out;
 }
+
+// Split a given absolute name into the parent model name and the local name.
+// If the give name is not scoped, this will return an empty string for the
+// parent model name and the given name as the local name.
+std::pair<std::string, std::string> SplitName(
+    const std::string &_absoluteName)
+{
+  const auto pos = _absoluteName.rfind(kSdfScopeDelimiter);
+  if (pos != std::string::npos)
+  {
+    const std::string first = _absoluteName.substr(0, pos);
+    const std::string second =
+        _absoluteName.substr(pos + kSdfScopeDelimiter.size());
+    return {first, second};
+  }
+  return {"", _absoluteName};
+}
+
+static bool EndsWithDelimiter(const std::string &_s)
+{
+  if (_s.size() < kSdfScopeDelimiter.size())
+    return false;
+
+  const size_t findStartPosition = _s.size() - kSdfScopeDelimiter.size();
+  return _s.substr(findStartPosition) == kSdfScopeDelimiter;
```

Comment:
nit: Probably more efficient to do `_s.find(kSdfScopeDelimiter, findStartPosition) == findStartPosition` as it avoids constructing a new string.

Reply:
Totally, switched to std::string::compare